eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -122,7 +129,7 @@ public function trash(Location $location)
             throw new InvalidArgumentValue('id', $location->id, 'Location');
         }
 
-        if (!$this->repository->canUser('content', 'remove', $location->getContentInfo(), [$location])) {
+        if (!$this->userHasPermissionsToRemove($location->getContentInfo(), $location)) {
```

Comment:
You could pass only a `$location`
```suggestion
        if (!$this->userHasPermissionsToRemove($location)) {
```